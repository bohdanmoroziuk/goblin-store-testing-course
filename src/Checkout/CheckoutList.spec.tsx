import React from 'react';
import { render } from '@testing-library/react';

import { Product } from "../shared/types";
import { CheckoutList } from "./CheckoutList";

describe("CheckoutList", () => {
  const products: Product[] = [
    {
      name: "Foo",
      price: 55,
      image: "/images/basic_spear.png",
    },
    {
      name: "Bar",
      price: 10,
      image: "/images/broad_sword.png",
    },
  ];

  it("renders list of products", () => {
    const { container } = render(<CheckoutList products={products} />)

    expect(container).toMatchSnapshot();
    expect(container.innerHTML).toMatch(products[0].name);
    expect(container.innerHTML).toMatch(products[1].name);
  })
})