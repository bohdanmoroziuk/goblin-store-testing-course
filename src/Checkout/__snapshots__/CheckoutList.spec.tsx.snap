// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutList renders list of products 1`] = `
<div>
  <ul
    class="nes-list is-circle"
  >
    <li>
      Foo
    </li>
    <li>
      Bar
    </li>
  </ul>
</div>
`;
